services/runservice/types/types.go

```diff
@@ -415,6 +415,7 @@ type RunStep struct {
 	Environment map[string]string `json:"environment,omitempty"`
 	WorkingDir  string            `json:"working_dir,omitempty"`
 	Shell       string            `json:"shell,omitempty"`
+	Tty         bool              `json:"tty,omitempty"`
```

Comment:
This will change the api of runs created before this change since tty will be false when unmarshalled. So it should be a `*bool` and Steps.UnmarshallJSON should default it to true if it's nil.

You should also add an Steps Unmarshal test to check that an input json RunConfig without a defined tty field should generate an output with tty set to true.